shell/platform/darwin/ios/rendering_api_selection.mm

```diff
@@ -21,26 +21,34 @@ bool ShouldUseMetalRenderer() {
   // past iOS 10.0. The processor was selected as it is the first version at which Metal was
   // supported. The iOS version floor was selected due to the availability of features used by Skia.
   bool ios_version_supports_metal = false;
+#if TARGET_IPHONE_SIMULATOR
+  if (@available(iOS 13.0, *)) {
+#else
   if (@available(iOS 10.0, *)) {
+#endif  // TARGET_IPHONE_SIMULATOR
     auto device = MTLCreateSystemDefaultDevice();
-    ios_version_supports_metal = [device supportsFeatureSet:MTLFeatureSet_iOS_GPUFamily1_v3];
+    // We need to check if the device is here since an ios 13 simulator running on an older version
+    // of macos (below 10.15) will not actually allow metal to be used.
+    if (device != nil) {
```

Comment:
This is not necessary in Objective-C. If device is `nil`, then sending a `supportsFeatureSet:` message to a `nil` object will return a zero-ed out return value (`NO`).

Reply:
Ah right, has been a while since I wrote any significant amount of objective C